glib2/ext/glib2/rbglib_regex.c

```diff
@@ -0,0 +1,513 @@
+/*
+ *  Copyright (C) 2015  Ruby-GNOME2 Project Team
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ *  MA  02110-1301  USA
+ */
+
+#include "rbgprivate.h"
+
+#ifdef G_OS_WIN32
+#include <windows.h>
+#endif
+
+#define RG_TARGET_NAMESPACE cRegex
+#define _SELF(s) ((GRegex*)RVAL2BOXED(s, G_TYPE_REGEX))
+
+static void
+regex_free(gpointer object)
+{
+    GRegex *regex = object;
+
+    g_regex_unref(regex);
+}
+
+static VALUE
+rg_initialize(gint argc, VALUE *argv, VALUE self)
+{
+    VALUE pattern, compile_options, match_options;
+    GError *error = NULL;
+    GRegex *regex = NULL;
+
+    rb_scan_args(argc, argv, "30", &pattern, &compile_options, &match_options);
+
+    regex = g_regex_new(RVAL2CSTR(pattern), 
+                        NUM2UINT(compile_options),
+                        NUM2UINT(match_options),
+                        &error);
+    
+    if(error)
+      RAISE_GERROR(error);
+
+    G_INITIALIZE(self, regex); 
+    return Qnil;
+}
+
+static VALUE
+rg_match_flags(VALUE self)
+{
+  return UINT2NUM(g_regex_get_match_flags(_SELF(self)));
+}
+
+static VALUE
+rg_compile_flags(VALUE self)
+{
+  return UINT2NUM(g_regex_get_compile_flags(_SELF(self)));
+}
+
+static VALUE
+rg_pattern(VALUE self)
+{
+  return CSTR2RVAL(g_regex_get_pattern(_SELF(self)));
+}
+
+static VALUE
+rg_max_backref(VALUE self)
+{
+  return INT2NUM(g_regex_get_max_backref(_SELF(self)));
+}
+
+static VALUE
+rg_capture_count(VALUE self)
+{
+  return INT2NUM(g_regex_get_capture_count(_SELF(self)));
+}
+
+static VALUE
+rg_has_cr_or_lf(VALUE self)
+{
+  gboolean contains_cr_or_lf = g_regex_get_has_cr_or_lf(_SELF(self));
+  
+  if(contains_cr_or_lf == TRUE)
+    return Qtrue;
+  else
+    return Qfalse;
+}
+
+static VALUE
+rg_max_lookbehind(VALUE self)
+{
+  return INT2NUM(g_regex_get_max_lookbehind(_SELF(self)));
+}
+
+static VALUE
+rg_string_number(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string;
+  rb_scan_args(argc, argv, "10", &string);
+  
+  return INT2NUM(g_regex_get_string_number(_SELF(self), RVAL2CSTR(string)));
+}
+
+static VALUE
+rg_split(gint argc, VALUE *argv, VALUE self)
+{
+  VALUE string, match_options, array_of_strings;
+  rb_scan_args(argc, argv, "20", &string, &match_options);
```

Comment:
Make `match_options` optional. And accept `start_position` and `max_tokens` to merge `rg_split_full`.

It's better to use optional argument in one method instead of multiple methods ends with `_full`. Because the former is Rubyish.